### ISSUE TITLE
Support deno

### ISSUE THREAD

**Issue by rarkins** [renovatebot/renovate#6237]
**What would you like Renovate to be able to do?**

Support Deno as a regular package manager. 

**Describe the solution you'd like**

Provide upgrades if someone uses branch versions

**Additional context**

https://deno.land/x/

**Comment by HonkingGoose** [comment]
@rarkins I saw you wanted to support the Deno datasource and manager on Slack. Turns out you already had a issue to add the Deno package manager. 😄 

Do you want a new issue that tracks Deno datasource support, or do you want to use the current issue for that?

You might want to increase the priority of this issue as well maybe?

**Comment by rarkins** [comment]
Yes, let's implement the datasource first, and it can be a separate issue

**Comment by secustor** [comment]
Blocked by https://github.com/renovatebot/renovate/issues/12724

**Comment by rarkins** [comment]
https://deno-blog.com/What_every_developer_should_know_about_the_Deno_third_party_module_registry.2023-04-09

**Comment by Hajime-san** [comment]
They are building official `jsr` registry👀

https://www.kitsonkelly.com/posts/jsr-first-impressions

**Comment by Hajime-san** [comment]
How about `jsr` datasource support?

**Comment by rarkins** [comment]
Please create a separate discussion (suggest an idea) with details such as the API spec, how it would work, etc.

**Comment by devsheva** [comment]
any news on this?

**Comment by 5ouma** [comment]
It's nice to be able to update `deno.lock` with JSR and npm.

**Comment by esimkowitz** [comment]
Why is there a reproduction tag on this? Is that affecting triage? What is required to get this going? I think Renovate has a great opportunity to become the go-to dependency lifecycle tool for Deno, since Dependabot doesn't seem to have any interest in it.

**Comment by Hajime-san** [comment]
Hi, I'm currently working on adding a Deno manager and a JSR datasource.
I think I can submit a PR in about 1 or 2 weeks.

**Comment by felipecrs** [comment]
That is great news, good luck to you, @Hajime-san
